src/vendor/components/navigation/Picker.jsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import MenuItem from '@material-ui/core/MenuItem';
+import TextField from '@material-ui/core/TextField';
+import { ArrayWrapper, selectFrom } from '../../vectors';
+import { missing } from '../../utils';
+
+const styles = () => ({
+  root: {
+    display: 'inline-block',
+    paddingTop: '2rem',
+    margin: '0 20px',
+  },
+});
+
+class Picker extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { value } = this.props;
+
+    this.state = { value };
```

Comment:
You may want to use `getDerivedStateFromProps` instead here. The danger here is that if the props on the component are changed without the component being recreated, the new prop will never be displayed because the constructor will never update the current state of the component. The initialization of state from props only runs when the component is first created.

Reply:
@helfi92 

I must think about that.  The intent is to set a default value for the Picker.  I can not foresee the `props` ever changing, and even if they did I can not imagine how that would happen, or what it would mean.  The Picker is meant to be the authority on the `value`; having another authority (presumably something that changes `props`) would be interesting.

Maybe the authority of `value` is the navigation component that holds all the Pickers; then the data flows from parent to children; then I must be concerned with `props` changing.

I have a question:  There appears to be two ways to deal with changing `props`: One is with `componentWillUpdate` and the other is ensure the parent sets the `key` attribute [example](https://github.com/mozilla-frontend-infra/firefox-health-dashboard/blob/e929da1c12e91e5ca97dddbc8ba3dea8b620b6ea/src/android/TP6m.jsx#L167).  The former only seems useful when there is no parent, or I do not want the parent to manage child state.  The latter is generally preferred because it lets me assume `props` will not change.  

You have some wise words about when either technique is preferred?

Thank you

Reply:
`componentWillUpdate` was actually [deprecated](https://reactjs.org/docs/react-component.html#unsafe_componentwillmount) in recent versions of React so I wouldn't recommend using them. The React docs has some [excellent documentation](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html) about when what to use. They also tackle the issue about when to use the `key` prop. Have a look and let me know if you still had questions. Happy to help.